paasta_tools/kubernetes/bin/paasta_cleanup_remote_run_resources.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python
+# Copyright 2015-2019 Yelp Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import argparse
+from datetime import datetime
+from datetime import timedelta
+from datetime import timezone
+from typing import Any
+from typing import Callable
+from typing import Sequence
+from typing import Tuple
+
+from paasta_tools.kubernetes.remote_run import get_remote_run_role_bindings
+from paasta_tools.kubernetes.remote_run import get_remote_run_roles
+from paasta_tools.kubernetes.remote_run import get_remote_run_service_accounts
+from paasta_tools.kubernetes_tools import get_all_managed_namespaces
+from paasta_tools.kubernetes_tools import KubeClient
+
+
+ListingFuncType = Callable[[KubeClient, str], Sequence[Any]]
+DeletionFuncType = Callable[[str, str], Any]
+
+
+def clean_namespace(kube_client: KubeClient, namespace: str, age_limit: datetime):
+    """Clean ephemeral remote-run resource in a namespace
+
+    :param KubeClient kube_client: kubernetes client
+    :param str namepsace: kubernetes namespace
+    :param datetime age_limit: expiration time for resources
+    """
+    cleanup_actions: Sequence[Tuple[DeletionFuncType, ListingFuncType]] = (
+        (
+            kube_client.core.delete_namespaced_service_account,
+            get_remote_run_service_accounts,
+        ),
+        (kube_client.rbac.delete_namespaced_role, get_remote_run_roles),
+        (kube_client.rbac.delete_namespaced_role_binding, get_remote_run_role_bindings),
+    )
+    for delete_func, list_func in cleanup_actions:
+        for entity in list_func(kube_client, namespace):
+            if (
+                not entity.metadata.name.startswith("remote-run-")
+                or entity.metadata.creation_timestamp > age_limit
```

Comment:
imo, we probably want to tag the resources with when they should be expired by - otherwise, we have to essentially always run this with `--max-age $REMOTE_RUN_JOB_MAX_DURATION` lest we delete an in-use SA/role/rolebinding

Reply:
We actually want to always delete these quickly. They should just exist for the user to be able to exec into the pod, once the shell session is established they can get removed.

Reply:
i guess there's not really an easy way for folks to exec back into the remote-run sandbox if their connection drops for whatever reason, so it does seem fine for this to happen :p